docs/user_manual/performance-guide.md

```diff
@@ -6,6 +6,57 @@ SPDX-License-Identifier: MPL-2.0
 
 # Guidelines for performance enhancement
 
+The `power-grid-model` is a library that shines in its ability to handle calculations at scale.
+It remains performant, even when doing calculations with one or a combination of the following extremes (non-exhaustive):
+
+- Large grids
+- Batch calculations with many scenarios
+- Many changes in the grid in each scenario
+
+To achieve that high performance, several optimizations to the code are made.
+To use those optimizations to their fullest extend, we recommend our users to follow the following guidelines.
+
+## Data validity
+
+Many of our optimizations rely on assuming input data validity and the fact that the provided grid is reasonably close to realistic.
+Non-convergence, underdetermined equations (sparse matrices) or other unexpected behavior may therefore be encountered when the data is not realistic.
+
+To keep the PGM performant, checks on hard physical bounds are offloaded to a separate tool, the [data validator](data-validator.md).
+However, these checks are extremely expensive and should therefore not be used in production environments at scale when performance matters.
```

Comment:
I believe this sentence is wrong in the wording, in production environments ppl also do debugging. Consider:
```suggestion
However, these checks can be prohibitively expensive and application at scale in production environments are therefore not recommended when performance matters.
```

Reply:
```suggestion
However, these checks can be prohibitively expensive and application at scale in production environments is therefore not recommended when performance matters.
```